src/components/AChat/AChat.vue

```diff
@@ -78,128 +78,110 @@ export default {
     }
   },
   emits: ['scroll', 'scroll:bottom', 'scroll:top'],
-  data: () => ({
-    currentScrollHeight: 0,
-    currentScrollTop: 0,
-    currentClientHeight: 0
-  }),
-  mounted() {
-    this.attachScrollListener()
-
-    this.currentClientHeight = this.$refs.messages.clientHeight
-    const resizeHandler = () => {
-      const clientHeightDelta = this.currentClientHeight - this.$refs.messages.clientHeight
-
-      const nonVisibleClientHeight =
-        this.$refs.messages.scrollHeight -
-        this.$refs.messages.clientHeight -
-        Math.ceil(this.$refs.messages.scrollTop)
-      const scrolledToBottom = nonVisibleClientHeight === 0
-
-      if (scrolledToBottom) {
-        // Browser updates Element.scrollTop by itself
-      } else {
-        this.$refs.messages.scrollTop += clientHeightDelta
-      }
-
-      this.currentClientHeight = this.$refs.messages.clientHeight
+  setup(props, { emit }) {
+    const messagesRef = ref<HTMLDivElement | null>(null)
+    const currentScrollHeight = ref(0)
+    const currentScrollTop = ref(0)
+    const currentClientHeight = ref(0)
+    const resizeObserver = ref<ResizeObserver | null>(null)
```

Comment:
You probably can initialize the `resizeObserver when creating a ref:

```ts
const resizeObserver = ref(new ResizeObserver(...))
```

So you don't need to check on `null` anymore.